### ISSUE TITLE
Prune issue - chunk can't be found

### ISSUE THREAD

**Issue by jt70471** [gilbertchen/duplicacy#182]
Hi @gilbertchen, I believe I may have found a bug related to prune processing. At the very end of the integration-tests/copy_test.sh script, add the following:

${DUPLICACY} prune -id integration-tests -r 1-2 -exhaustive -exclusive -storage default

This should remove the two revisions made by the two backup commands. The attached output shows that indeed snapshot revisions 1 and 2 are deleted, but then after it would seem that prune should be finished, the oddball "can't be found" message is shown. Here's the snippet:

```
The snapshot integration-tests at revision 1 has been removed
The snapshot integration-tests at revision 2 has been removed
File /Users/jefft/DUPLICACY_TEST_ZONE/TEST_STORAGE/chunks/b3/1c/60e7eaca42d36488b50281549c42d0082034640b07fc975d4a5b91a5184c can't be found: lstat /Users/jefft/DUPLICACY_TEST_ZONE/TEST_STORAGE/chunks/b3/1c/60e7eaca42d36488b50281549c42d0082034640b07fc975d4a5b91a5184c: no such file or directory
File /Users/jefft/DUPLICACY_TEST_ZONE/TEST_STORAGE/chunks/b3/1c/60e7eaca42d36488b50281549c42d0082034640b07fc975d4a5b91a5184c.fsl can't be found: lstat /Users/jefft/DUPLICACY_TEST_ZONE/TEST_STORAGE/chunks/b3/1c/60e7eaca42d36488b50281549c42d0082034640b07fc975d4a5b91a5184c.fsl: no such file or directory
Chunk b31c60e7eaca42d36488b50281549c42d0082034640b07fc975d4a5b91a5184c can't be found
```
The chunk is in both snapshot revisions, so makes sense that once both revisions are removed, the chunk should have been deleted.

The only place I see this exact message being issued is in duplicacy_chunkdownloader.go, which makes no sense given that this was a prune command, not sure why this chunk would attempt to be downloaded, etc.

I've started looking into it, but I figured I'd report it as I'm sure you are much more familiar with why this may be occurring.

Any questions, please let me know. Thanks. The full output of the prune command is attached 
[here](https://github.com/gilbertchen/duplicacy/files/1293419/a.txt).

**Comment by gilbertchen** [comment]
I think I know what happened here.  CleanSnapshotCache takes the latest snapshot as the first argument because it needs to keep snapshot chunks referenced by the last backup.  The latest snapshot can be nil, but if it is not nil then Duplicacy will need to find out which snapshot chunks are needed by downloading chunks that contain this information.  However, all chunks have been removed from the chunk therefore it reports a chunk is missing. 

So the bug is that when all snapshots are deleted, the latest snapshot passed to CleanSnapshotCache should be nil.

**Comment by jt70471** [comment]
I have worked something up that prevents the error, but not entirely sure I've addressed it correctly. Please take a look at the PR and let me know if you had envisioned something different.

**Comment by jt70471** [comment]
This can be closed since PR #183 has been merged.
